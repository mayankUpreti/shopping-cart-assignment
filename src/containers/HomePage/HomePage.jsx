import React,{useState,useEffect} from 'react'
import CustomCarousel from '../../components/Carousel/Carousel'
import {fetchbanners, fetchcategories} from '../../api/api'
import CategoryCard from '../../components/CategoryCard/CategoryCard'

const HomePage=()=>{

    const [banners,setBanners]=useState([])
    const [categories,setCategories]=useState([])
    useEffect(async()=>{
        const data=await fetchbanners();
        setBanners(data);
        const category=await fetchcategories();
        setCategories(category)
    },[])

return(
    <div className='main-body '>
       <CustomCarousel data={banners}/>
       {
           categories && categories.map((el,i)=>
           el.enabled &&
           <CategoryCard key={el.key} data={el} order={i}/>)
       }
    
    </div>
)

}

export default HomePage