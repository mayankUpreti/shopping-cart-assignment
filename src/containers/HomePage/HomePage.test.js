import React  from 'react'
import HomePage from './HomePage'
import {shallow} from 'enzyme'


it('testing homepage ',()=>{

    expect(shallow(<HomePage  />)).toMatchSnapshot()

})