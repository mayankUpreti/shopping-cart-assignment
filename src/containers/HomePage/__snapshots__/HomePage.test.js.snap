// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing homepage  1`] = `
<div
  className="main-body "
>
  <CustomCarousel
    data={Array []}
  />
</div>
`;
